File: app/src/main/java/com/drdisagree/iconify/services/TileMonetEngine.java
package com.drdisagree.iconify.services;

import static com.drdisagree.iconify.common.Preferences.MONET_ENGINE_SWITCH;

import android.service.quicksettings.Tile;
import android.service.quicksettings.TileService;

import com.drdisagree.iconify.R;
import com.drdisagree.iconify.config.Prefs;
import com.drdisagree.iconify.utils.OverlayUtil;

public class TileMonetEngine extends TileService {

    private boolean isCustomMonetEnabled = Prefs.getBoolean("IconifyComponentME.overlay");

    @Override
    public void onTileAdded() {
        super.onTileAdded();
    }

    @Override
    public void onStartListening() {
        super.onStartListening();
        Tile customMonetTile = getQsTile();
        customMonetTile.setState(isCustomMonetEnabled ? Tile.STATE_ACTIVE : Tile.STATE_INACTIVE);
        customMonetTile.updateTile();
    }

    @Override
    public void onStopListening() {
        super.onStopListening();
    }

    @Override
    public void onClick() {
        super.onClick();

        if (isCustomMonetEnabled) {
            OverlayUtil.disableOverlays("IconifyComponentDM.overlay", "IconifyComponentME.overlay");
        } else {
            OverlayUtil.enableOverlays("IconifyComponentDM.overlay", "IconifyComponentME.overlay");

            if (Prefs.getBoolean("IconifyComponentQSPBD.overlay")) {
                OverlayUtil.changeOverlayState("IconifyComponentQSPBD.overlay", false, "IconifyComponentQSPBD.overlay", true);
            } else if (Prefs.getBoolean("IconifyComponentQSPBA.overlay")) {
                OverlayUtil.changeOverlayState("IconifyComponentQSPBA.overlay", false, "IconifyComponentQSPBA.overlay", true);
            }
        }

        isCustomMonetEnabled = !isCustomMonetEnabled;
        Prefs.putBoolean(MONET_ENGINE_SWITCH, isCustomMonetEnabled);

        Tile customMonetTile = getQsTile();
        customMonetTile.setState(isCustomMonetEnabled ? Tile.STATE_ACTIVE : Tile.STATE_INACTIVE);
        customMonetTile.setLabel(getResources().getString(R.string.activity_title_monet_engine));
        customMonetTile.setContentDescription(isCustomMonetEnabled ? getResources().getString(R.string.general_on) : getResources().getString(R.string.general_off));
        customMonetTile.updateTile();
    }

    @Override
    public void onTileRemoved() {
        super.onTileRemoved();
    }
}

File: app/src/main/java/com/drdisagree/iconify/services/TileMonetEngine.kt
package com.drdisagree.iconify.services

import android.service.quicksettings.Tile
import android.service.quicksettings.TileService
import com.drdisagree.iconify.R
import com.drdisagree.iconify.common.Preferences.MONET_ENGINE_SWITCH
import com.drdisagree.iconify.config.RPrefs.getBoolean
import com.drdisagree.iconify.config.RPrefs.putBoolean
import com.drdisagree.iconify.utils.overlay.OverlayUtils


class TileMonetEngine : TileService() {

    private var isCustomMonetEnabled = getBoolean("IconifyComponentME.overlay")

    override fun onStartListening() {
        super.onStartListening()

        val tile = qsTile
        tile.state = if (isCustomMonetEnabled) Tile.STATE_ACTIVE else Tile.STATE_INACTIVE
        tile.updateTile()
    }

    override fun onClick() {
        super.onClick()

        if (isCustomMonetEnabled) {
            OverlayUtils.disableOverlays("IconifyComponentDM.overlay", "IconifyComponentME.overlay")
        } else {
            OverlayUtils.enableOverlays("IconifyComponentDM.overlay", "IconifyComponentME.overlay")

            if (getBoolean("IconifyComponentQSPBD.overlay")) {
                OverlayUtils.changeOverlayState(
                    "IconifyComponentQSPBD.overlay",
                    false,
                    "IconifyComponentQSPBD.overlay",
                    true
                )
            } else if (getBoolean("IconifyComponentQSPBA.overlay")) {
                OverlayUtils.changeOverlayState(
                    "IconifyComponentQSPBA.overlay",
                    false,
                    "IconifyComponentQSPBA.overlay",
                    true
                )
            }
        }

        isCustomMonetEnabled = !isCustomMonetEnabled
        putBoolean(MONET_ENGINE_SWITCH, isCustomMonetEnabled)

        val tile = qsTile
        tile.state = if (isCustomMonetEnabled) Tile.STATE_ACTIVE else Tile.STATE_INACTIVE
        tile.label = resources.getString(R.string.activity_title_monet_engine)
        tile.subtitle =
            if (isCustomMonetEnabled) resources.getString(R.string.general_on)
            else resources.getString(R.string.general_off)

        tile.updateTile()
    }
}
